src/control/src/node_/node_control.cpp

```diff
@@ -103,4 +108,40 @@ void Control::publish_evaluator_data(double lookahead_velocity, Position lookahe
   evaluator_data.closest_point.y = closest_point.y;
   evaluator_data.lookahead_velocity = lookahead_velocity;
   this->evaluator_data_pub_->publish(evaluator_data);
+}
+
+void Control::publish_visualization_data(const Position& lookahead_point,
+                                         const Position& closest_point) const {
+  // Visualization
+  // Publish the lookahead point and the closest point
+  // to visualize the control
+  auto marker = visualization_msgs::msg::Marker();
+  marker.header.frame_id = "map";
+  marker.header.stamp = this->now();
+  marker.ns = "control";
+  marker.id = 0;
+  marker.type = visualization_msgs::msg::Marker::SPHERE;
+  marker.action = visualization_msgs::msg::Marker::ADD;
+  marker.pose.position.x = lookahead_point.x;
+  marker.pose.position.y = lookahead_point.y;
+  marker.pose.position.z = 0.1;
+  marker.pose.orientation.x = 0.0;
+  marker.pose.orientation.y = 0.0;
+  marker.pose.orientation.z = 0.0;
+  marker.pose.orientation.w = 1.0;
+  marker.scale.x = 0.2;
+  marker.scale.y = 0.2;
+  marker.scale.z = 0.2;
+  marker.color.a = 1.0;  // Don't forget to set the alpha!
+  marker.color.r = 0.0;
+  marker.color.g = 1.0;
+  marker.color.b = 0.0;
+  this->closest_point_pub_->publish(marker);
+
+  marker.id = 1;
+  marker.pose.position.x = closest_point.x;
+  marker.pose.position.y = closest_point.y;
+  marker.color.r = 1.0;
+  marker.color.g = 0.0;
```

Comment:
I know the common_lib function can't handle this message. Is there a way to correct it?